BookClubSocialNetwork/urls.py

```diff
@@ -18,24 +18,41 @@
 from django.urls import path
 from BookClub import views
 
+# from BookClub.views.action_views import DemoteMemberView, PromoteView
+
 urlpatterns = [
     # '''Core URLs'''
     path('admin/', admin.site.urls),
     path('', views.home, name='home'),
 
+
     # '''Authentication URLs'''
     path('login/', views.LogInView.as_view(), name='login'),
     path('sign_up/', views.SignUpView.as_view(), name='sign_up'),
     path('log_out/', views.LogOutView.as_view(), name='log_out'),
-    
     # '''User URLs'''
     path('user_dashboard/', views.UserDashboardView.as_view(), name='user_dashboard'),
     path('edit_profile/', views.EditProfileView.as_view(), name='edit_profile'),
     path('password_change/', views.ChangePasswordView.as_view(), name='password_change'),
     
+    # '''Action URLs'''
+    path('join_club/<str:url_name>/', views.JoinClubView.as_view(), name='join_club'),
+    path('leave_club/<str:url_name>/', views.LeaveClubView.as_view(), name='leave_club'),
+    path('promote_member/<str:url_name>/', views.PromoteMemberView.as_view(), name='promote_member'),
+    path('demote_member/<str:url_name>/', views.DemoteMemberView.as_view(), name='demote_member'),
+    path('transfer_ownership/<str:url_name>/', views.TransferOwnershipView.as_view(), name='transfer_ownership'),
+    path('kick_member/<str:url_name>/', views.KickMemberView.as_view(), name='kick_member'),
+    path('delete_club/<str:url_name>/',views.DeleteClubView.as_view(),name='delete_club'),
+
+    # '''Membership URLs'''
+    path('club/available_clubs/', views.AvailableClubsView.as_view(), name='available_clubs'),
+    path('club/my_club_memberships/', views.MyClubMembershipsView.as_view(), name='my_club_memberships'),
+    path('club/create_club/', views.clubs.CreateClubView.as_view(), name = 'create_club'),
+
     # '''Club URLs'''
-    path('club_dashboard/', views.club_dashboard, name='club_dashboard'), 
-    path('available_clubs/', views.AvailableClubsView.as_view(), name='available_clubs'),
-    path('my_club_memberships/', views.MyClubMembershipsView.as_view(), name='my_club_memberships'),
-    path('create_club/', views.clubs.CreateClubView.as_view(), name = 'create_club')
+    path('club/<str:club_url_name>/', views.ClubDashboardView.as_view(), name='club_dashboard'),
+    path('club/<str:url_name>/member_list/', views.MembersListView.as_view(), name='member_list'),
```

Comment:
we need to standardize the name of the URL variable. I suggest club_url_name for greater clarity in naming.

Reply:
as the model uses url_name, we should stick to url_name

Reply:
then we should update the model instead, I think it's a better variable name as you know what the url_name is representing.